src/ORM/DataList.php

```
@@ -308,72 +313,106 @@ public function distinct($value)
     }
 
     /**
-     * Return a new DataList instance as a copy of this data list with the sort
-     * order set.
+     * Return a new DataList instance as a copy of this data list with the sort order set
      *
-     * @see SS_List::sort()
-     * @see SQLSelect::orderby
+     * Raw SQL is not accepted, only actual field names can be passed
+     *
+     * @param string|array $args
      * @example $list = $list->sort('Name'); // default ASC sorting
-     * @example $list = $list->sort('Name DESC'); // DESC sorting
+     * @example $list = $list->sort('Name ASC, Age DESC');
      * @example $list = $list->sort('Name', 'ASC');
-     * @example $list = $list->sort(array('Name'=>'ASC', 'Age'=>'DESC'));
-     *
-     * @param string|array Escaped SQL statement. If passed as array, all keys and values are assumed to be escaped.
-     * @return static
+     * @example $list = $list->sort(['Name' => 'ASC', 'Age' => 'DESC']);
      */
-    public function sort()
+    public function sort(...$args): static
     {
-        $count = func_num_args();
-
+        $count = count($args);
         if ($count == 0) {
             return $this;
         }
-
         if ($count > 2) {
             throw new InvalidArgumentException('This method takes zero, one or two arguments');
         }
-
         if ($count == 2) {
-            $col = null;
-            $dir = null;
-            list($col, $dir) = func_get_args();
-
-            // Validate direction
-            if (!in_array(strtolower($dir ?? ''), ['desc', 'asc'])) {
-                user_error('Second argument to sort must be either ASC or DESC');
-            }
```

Why did you remove this validation? What if someone calls the method like this:
```php
$list = $list->sort('Name ASC', 'Age DESC');
```
This should should throw an `InvalidArgumentException` I think (and the change from `user_error` to `InvalidArgumentException` might need to be mentioned in the changelog)

---

I was originally validating the sort just above the calls to `$this->validateColumnHasNoRawSql($column);` - though it's not really required as exceptions are thrown regardless, just at a later stage.  We have unit testing for this

---

Have added direction validation back in